spring-cloud-starter-stream-processor-python-http/src/main/java/org/springframework/cloud/stream/app/python/http/processor/PythonHttpProcessorConfiguration.java

```diff
@@ -16,26 +16,69 @@
 
 package org.springframework.cloud.stream.app.python.http.processor;
 
-import org.springframework.cloud.stream.aggregate.AggregateApplication;
-import org.springframework.cloud.stream.aggregate.AggregateApplicationBuilder;
-import org.springframework.cloud.stream.app.httpclient.processor.HttpclientProcessorConfiguration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.app.httpclient.processor.HttpclientProcessorFunctionConfiguration;
+import org.springframework.cloud.stream.app.python.jython.JythonScriptExecutor;
 import org.springframework.cloud.stream.app.python.wrapper.JythonWrapperConfiguration;
+import org.springframework.cloud.stream.messaging.Processor;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
 
 /**
  * @author David Turanski
  **/
 @Configuration
-@Import(JythonWrapperConfiguration.class)
+@Import({ JythonWrapperConfiguration.class, HttpclientProcessorFunctionConfiguration.class })
+@EnableBinding(Processor.class)
 public class PythonHttpProcessorConfiguration {
+
+	@Autowired(required = false)
+	private JythonScriptExecutor jythonWrapper;
+
+	@Bean
+	IntegrationFlow pythonHttpFlow(Processor processor,
+		Function<Message<?>, Message<?>> preProcess,
+		@Qualifier(HttpclientProcessorFunctionConfiguration.HTTPCLIENT_PROCESSOR_FUNCTION_NAME)
+			Function<Message<?>, Object> httpRequest,
+		Function<Message<?>, Message<?>> postProcess) {
+		return IntegrationFlows.from(processor.input())
+			.transform(Message.class, preProcess::apply)
+			.transform(Message.class, httpRequest::apply)
+			.transform(Message.class, postProcess::apply)
+			.channel(processor.output())
+			.get();
+
+	}
+
 	@Bean
-	public AggregateApplication pythonProcessorApp() {
-		return new AggregateApplicationBuilder().parent(JythonWrapperConfiguration.class)
-				.from(HttpJythonProcessorInputConfiguration.class).namespace("in")
-				.via(HttpclientProcessorConfiguration.class)
-				.to(HttpJythonProcessorOutputConfiguration.class)
-				.namespace("out").build();
+	public Function<Message<?>, Message<?>> preProcess() {
+		return message -> applyJythonWrapper(message, Processor.INPUT);
+	}
+
+	@Bean
+	public Function<Message<?>, Message<?>> postProcess() {
```

Comment:
Same as above (to use custom name for the function bean).

Reply:
For `preProcess` an `postProcess` here, I'm not too worried about collisions, since there is not one already. If a user would extend this app and add a function of the same name it would be on them.